lib/SearchForm/SearchForm.test.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import { noop } from 'lodash';
+
+import {
+  screen, render,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import '../../test/jest/__mock__';
+import '../../test/jest/__new_mock__';
+
+import { SearchForm } from './SearchForm';
+
+const setupSearchForm = ({
+  searchTerm = 'Search term',
+  contextName = 'test',
+  searchLabelKey = 'test',
+  isLoading = false,
+  idKey,
+  handleChange,
+  handleSubmit,
+}) => {
+  render(
+    <SearchForm
+      searchTerm={searchTerm}
+      contextName={contextName}
+      searchLabelKey={searchLabelKey}
+      isLoading={isLoading}
+      idKey={idKey}
+      handleChange={handleChange}
+      handleSubmit={e => {
+        e.preventDefault();
+        handleSubmit();
+      }}
+      handleClear={noop}
+    />
+  );
+};
+
+describe('SearchForm', () => {
+  describe('rendering search form with non empty search term', () => {
+    let searchTerm = 'Search term';
+    const handleChangeMock = jest.fn();
+    const handleSubmitMock = jest.fn();
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should display search form', () => {
+      setupSearchForm({
+        searchTerm,
+        handleChange: handleChangeMock,
+        handleSubmit: handleSubmitMock,
+      });
+
+      expect(document.querySelector('[data-test-search-form]')).toBeVisible();
+    });
+    it('should display search form field with correct value', () => {
+      setupSearchForm({
+        searchTerm,
+        handleChange: handleChangeMock,
+        handleSubmit: handleSubmitMock,
+      });
+
+      const searchBox = screen.getByRole('searchbox');
+
+      expect(searchBox).toBeVisible();
+      expect(searchBox).toHaveValue(searchTerm);
+    });
+
+    it('should display search form submit button', () => {
+      setupSearchForm({
+        searchTerm,
+        handleChange: handleChangeMock,
+        handleSubmit: handleSubmitMock,
+      });
+
+      expect(screen.getByRole('button', { name: 'stripes-smart-components.search' })).toBeVisible();
```

Comment:
ideally they should be 2 different cases

Reply:
> ideally they should be 2 different cases

Ye, I'm testing enable or disable button in cases below.

Reply:
why do you check that button is visible when you test enable/disable?

Reply:
Just check if it present on the page first of all, to be sure in that fact. And after that I check it enable or not. In other cases I can be sure, that button visible on the page, so I can test on enable/disable.